### DISCUSSION TITLE
v0.6.2 -> v0.7.2. How to handle mixed clients?

### DISCUSSION THREAD

**Discussion by tommy-uytterhaegen** [Mimetis/Dotmim.Sync#507] [Q&A]
I currently have my clients on v0.6.2 and want to upgrade to v0.7.2.

How should I proceed with this? I can push an app update (currently only Android), but users typically don't upgrade the second you release :) 
This means that there will be a point where some clients will still be at 0.6.2 while others are at 0.7.2. The server would also be updated to 0.7.2 (while some clients are at 0.6.2). Is this a problem? Even if it is not for the current release. How should we handle different versions? Or will DMS always be backwards compatible?

**Comment by Mimetis** [comment]
Both Versions are compatible.
You can have clients on version `v0.6.2` and server on `v0.7.2`
Be careful, I did not test the reverse (clients on `v0.7.2` and server still on `v0.6.2` )

So a good way to update everything would be to upgrade your server, see if everything is still working fine, and then slowly upgrade your clients.


To upgrade a client or server, the `UpgradeAsync` method accessible from your orchestrator will allow you to upgrade to the last version.
That being said, notice that:
- **Client Side** : No need to call `UpgradeAsync`, since it's handled by default by `DMS`. On the next Sync, the method will be automatically called by `DMS`.
- **Server Side** : You need to create a `RemoteOrchestrator` with the correct `SyncSetup` and call the `UpgradeAsync` method. My advice would be to create a small console application and run the code once. Once it's done, no need to re execute it.

``` cs
// Upgrade to last version
if (await remoteOrchestrator.NeedsToUpgradeAsync())
   await remoteOrchestrator.UpgradeAsync(progress:progress);
```

Once upgraded your schema should be updated to the last version

Here is a screenshot (pretty old) where a schema has been upgraded to `v0.6.2`  (in your situation, you may have `v0.7.2`):
![image](https://user-images.githubusercontent.com/4592555/108779285-ebe99580-7566-11eb-89ea-1b77c7c88c3a.png)

**Reply by tommy-uytterhaegen** [reply]
Thank you for the swift reply!
If interested. I've implemented your library in My Board Game Collection, [open beta on Android](http://myboardgamecollection.net/) (testing out DMS :)), but the releases are being readied for iOS & Windows 10 as well.

**Reply by Mimetis** [reply]
Hey @Tuytje.
I saw your *My Board Game Collection*. this application seems very promising !

I was wondering... Would you be interested in sharing what you've done with `DMS` and your application using it, in a special dedicated page in the official `DMS` documentation https://dotmimsync.readthedocs.io/ ?

I mean, I'm considering to add a new section to the doc (just after `Overview` and before `How does it work`) called something like `They are using it` which will contains testimonials from people using `DMS` in their application / company.

The only thing I need from you is a markdown file describing your project, with a feedback on `DMS`. Maybe one or 2 images, code, links; whatever :)
It could be a simple paragraph, or a full page, you can choose the format and size your prefer.

Interested ?

cc @vunder @VagueGit, @maxrodro, @sylverbee, @gentledepp, @workgroupengineering; @gb0o

**Reply by sylverbee** [reply]
Hi Mimetis. Very nice idea.
Here is my app (Virtual Canvas for W10) and its link : [https://www.microsoft.com/en-us/p/virtual-canvas/9nblggh4td3g?activetab=pivot:overviewtab](url)

**Reply by maxrodro** [reply]
Hello everyone, I am going to start a project with the DMS, in the tests it has been very successful.

When I have it ready I will put my testimony.

**Reply by tommy-uytterhaegen** [reply]
> Hey @Tuytje.
> I saw your _My Board Game Collection_. this application seems very promising !
> 
> I was wondering... Would you be interested in sharing what you've done with `DMS` and your application using it, in a special dedicated page in the official `DMS` documentation https://dotmimsync.readthedocs.io/ ?
> 
> I mean, I'm considering to add a new section to the doc (just after `Overview` and before `How does it work`) called something like `They are using it` which will contains testimonials from people using `DMS` in their application / company.
> 
> The only thing I need from you is a markdown file describing your project, with a feedback on `DMS`. Maybe one or 2 images, code, links; whatever :)
> It could be a simple paragraph, or a full page, you can choose the format and size your prefer.
> 
> Interested ?
> 
> cc @vunder @VagueGit, @maxrodro, @sylverbee, @gentledepp, @workgroupengineering; @gb0o

Hey @Mimetis, I'd happy to help with that. Or at least do my best, documentation (even if it is based on experiences) is hard :D Do you need me to write it directly on the page, or send you via pm?

**Reply by VagueGit** [reply]
I have created a discussion for [Testimonials](https://github.com/Mimetis/Dotmim.Sync/discussions/514). It would be better if @Mimetis took ownership?

**Reply by Mimetis** [reply]
Thanks @VagueGit ! It's a good idea :)
I've created the [Testimonials ](https://github.com/Mimetis/Dotmim.Sync/discussions/516) discussion. I think it's better if the discussion is starting from the author's message, to be sure there is no confusion for new people discovering the framework.
(I've deleted your new post; sorry ! ;))

We can continue this [discussion on this new thread](https://github.com/Mimetis/Dotmim.Sync/discussions/516).
Can wait to see what you've done with `DMS` !!!! (I'm SO Fu... happy !)
